src/libs/antares/study/parts/thermal/cluster_list.cpp

```diff
@@ -606,18 +585,31 @@ bool ThermalClusterList::saveEconomicCosts(const AnyString& folder) const
 
 bool ThermalClusterList::loadPreproFromFolder(Study& study, const AnyString& folder)
 {
-    const bool globalThermalTSgeneration = study.parameters.timeSeriesToGenerate
-                                           & timeSeriesThermal;
-
     Clob buffer;
     auto hasPrepro = [](auto c) { return (bool)c->prepro; };
 
-    auto loadAndCheckPrepro = [&buffer, &folder, &study, &globalThermalTSgeneration](auto c)
+    auto loadPrepro = [&buffer, &folder, &study](auto& c)
     {
         assert(c->parentArea && "cluster: invalid parent area");
         buffer.clear() << folder << SEP << c->parentArea->id << SEP << c->id();
 
-        bool result = c->prepro->loadFromFolder(study, buffer);
+        return c->prepro->loadFromFolder(study, buffer);
+    };
+
+    return std::ranges::all_of(allClusters_ | std::views::filter(hasPrepro), loadPrepro);
+}
+
+bool ThermalClusterList::validatePrepro(const Study& study)
+{
+    auto hasPrepro = [](auto c) { return (bool)c->prepro; };
+
+    const bool globalThermalTSgeneration = study.parameters.timeSeriesToGenerate
+                                           & timeSeriesThermal;
+
+    return std::ranges::all_of(allClusters_ | std::views::filter(hasPrepro),
+            [&study, &globalThermalTSgeneration](auto& c)
+    {
+        bool result = true;
```

Comment:
if NOT (study.usedByTheSolver && globalThermalTSgeneration)  that is to say if !study.usedByTheSolver or !globalThermalTSgeneration result will be true. Moreover it is constant for all element.
It seems that we can return true before the computation of all_of if either value is false.